v4/key_test.go

```diff
@@ -18,6 +19,53 @@ import (
 	"github.com/miekg/pkcs11"
 )
 
+func init() {
+	var err error
+	ecPubKey, err = genECPubKey()
+	if err != nil {
+		log.Fatal(err)
+	}
+	ecPoint, err = getECPoint(ecPubKey.(*ecdsa.PublicKey))
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// Generate an ephemeral ECDSA key pair and return the public key corresponding
+// to the private key. If a key pair cannot be generated, an error is returned.
+func genECPubKey() (crypto.PublicKey, error) {
+	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	return privateKey.Public(), nil
+}
+
+// Retrieve the DER-encoding of ANSI X9.62 ECPoint value 'Q' from a given ECDSA
+// public key to match against a PKCS#11 CKA_EC_POINT attribute. If the
+// point cannot be marshalled, an error will be returned.
+//
+// [1]:
+// http://docs.oasis-open.org/pkcs11/pkcs11-curr/v2.40/cos01/pkcs11-curr-v2.40-cos01.html#_Toc408226932
+// [2]:
+// https://www.itu.int/ITU-T/formal-language/itu-t/x/x894/2018-cor1/ANSI-X9-62.html
```

Comment:
nit: when the URLs are going to be longer than the line width limit _anyway_, might as well put them on the same line as their footnote reference just for readability.
```suggestion
// [1]: http://docs.oasis-open.org/pkcs11/pkcs11-curr/v2.40/cos01/pkcs11-curr-v2.40-cos01.html#_Toc408226932
// [2]: https://www.itu.int/ITU-T/formal-language/itu-t/x/x894/2018-cor1/ANSI-X9-62.html
```

I'm not sure what paragraph is referencing these footnotes, though, and there seem to be some sentence fragments on lines 53 and 55? Maybe pull those sentence fragments out into their own straight-line sentence with references down to these URLs?